libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -230,14 +205,72 @@ export function OutQueueManager(
     return typeof queue[0] === 'object' && 'evt' in queue[0];
   };
 
+  // Runs `onRequestFailure` callback if defined
+  const tryOnRequestFailure = (status: number, message: string, events: EventBatch, willRetry: boolean) => {
+    onRequestFailure?.({ events, status, message, willRetry });
+  };
+
   /**
    * Send event as POST request right away without going to queue. Used when the request surpasses maxGetBytes or maxPostBytes
    * @param body POST request body
    * @param configCollectorUrl full collector URL with path
    */
   function sendPostRequestWithoutQueueing(body: PostEvent, configCollectorUrl: string) {
     const xhr = initializeXMLHttpRequest(configCollectorUrl, true, false);
-    xhr.send(encloseInPayloadDataEnvelope(attachStmToEvent([body.evt])));
+    const batch = attachStmToEvent([body.evt]);
+
+    xhr.onreadystatechange = function () {
+      if (xhr.readyState === 4) {
+        if (isSuccessfulRequest(xhr.status)) {
+          onRequestSuccess?.(batch);
+        } else {
+          tryOnRequestFailure(xhr.status, xhr.statusText, batch, false);
+        }
+      }
+    };
+
+    xhr.send(encloseInPayloadDataEnvelope(batch));
+  }
+
+  const removeEventsFromQueue = (numberToSend: number): void => {
```

Comment:
Nit: Maybe use the `function ... () { .. }` syntax to match with the other functions here?

Reply:
Good point, this was moved from [here](https://github.com/snowplow/snowplow-javascript-tracker/pull/1263/files/f2b574fca59a1424e1fb361ef08b2a4fa51e3aea#diff-a57c80855fac59c2b4a6266d84fed6d821cfba1991cbfecc49924a515c54e942L363), since I didn't want to define even more functions in `exeuteQueue` and needed it in scope for `setXhrCallbacks`. Now it's defined outside a function we can use `function ... ()` 😄